sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubServiceClient.cs

```diff
@@ -0,0 +1,969 @@
+        /// <param name="permission">Current supported actions are joinLeaveGroup and sendToGroup.</param>
+        /// <param name="connectionId">Target connection Id.</param>
+        /// <param name="targetName">If not set, get the permission for all targets. If set, get the permission for the specific target. The meaning of the target depends on the specific permission.</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        public virtual async Task<Response<bool>> CheckPermissionAsync(WebPubSubPermission permission, string connectionId, string targetName = default, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope("WebPubSubServiceRestClient.CheckPermissionAsync");
+            scope.Start();
+            try
+            {
+                var response = await RestClient.CheckPermissionAsync(hub, ToPermission(permission), connectionId, targetName, cancellationToken).ConfigureAwait(false);
+                return Response.FromValue(response.Status == 200, response);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>Check if a connection have permission to the specific action.</summary>
+        /// <param name="permission">Body of the requerst. Either `joinLeaveGroup` or `sendToGroup`.</param>
+        /// <param name="connectionId">Target connection Id.</param>
+        /// <param name="targetName">If not set, get the permission for all targets. If set, get the permission for the specific target. The meaning of the target depends on the specific permission.</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>A <see cref="Response"/> representing the result of the operation.</returns>
+        public virtual Response<bool> CheckPermission(WebPubSubPermission permission, string connectionId, string targetName = default, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope("WebPubSubServiceRestClient.CheckPermission");
+            scope.Start();
+            try
+            {
+                var response = RestClient.CheckPermission(hub, ToPermission(permission), connectionId, targetName, cancellationToken);
+                return Response.FromValue(response.Status == 200, response);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Close the client connection.
+        /// </summary>
+        /// <param name="connectionId">Target connection Id.</param>
+        /// <param name="reason">The reason closing the client connection.</param>
```

Comment:
Could we mention what the reason is used for/how it's exposed?  It would tell me more about what kind of value I want to pass here.